app/src/main/java/com/kickstarter/libs/htmlparser/ElementExt.kt

```diff
@@ -0,0 +1,55 @@
+package com.kickstarter.libs.htmlparser
+
+import org.jsoup.nodes.Element
+
+fun Element.extractViewElementTypeFromDiv(): ViewElementType {
+    var type: ViewElementType = ViewElementType.UNKNOWN
+
+    if (this.isImageStructure()) {
+        if (this.children().getOrNull(0)?.children()?.getOrNull(0)?.tag()?.name == ViewElementType.IMAGE.tag) {
+            type = ViewElementType.IMAGE
+        }
+    } else if (this.isIframeStructure()) {
+        if (this.children().getOrNull(0)?.tag()?.name == ViewElementType.EXTERNAL_SOURCES.tag) {
+            type = ViewElementType.EXTERNAL_SOURCES
+        }
+    }
+
+    return type
+}
+
+fun Element.isIframeStructure(): Boolean {
+    val isTemplateDiv = this.attributes().filter {
+        it.key == "class" && it.value == "template oembed"
+    }
+
+    return !isTemplateDiv.isNullOrEmpty()
+}
+
+fun Element.isImageStructure(): Boolean {
+    val isTemplateDiv = this.attributes().filter {
+        it.key == "class" && it.value == "template asset"
+    }
+
+    return !isTemplateDiv.isNullOrEmpty()
+}
+
+fun Element.parseImageElement(): ImageViewElement {
+    var src = ""
+    var caption: String? = null
+    var href: String? = null
+
+    if (this.parent()?.tag()?.name == "a") {
+        href = this.parent()?.attr("href")
+    }
+
+    caption = this.attr("data-caption")
+    src = this.children().getOrNull(0)?.children()?.getOrNull(0)?.attr("src").toString()
```

Comment:
For gif images the URL is in that `data-src=$data_src ` we need to get the gif  url from data-src
```
if (element.hasAttr("data-src")) {
    element.attributes()["data-src"].apply {
        viewElements.add(ImageViewElement(this))
    }
} else {
    element.attributes()["src"].apply {
        viewElements.add(ImageViewElement(this))
    }
}
```
![image](https://user-images.githubusercontent.com/1075310/141125241-83ac3cda-3d3f-4661-b5b5-b553d41cb83b.png)

Reply:
Done! line 50